src/Templates/ODataT4CodeGenerator.cs

```diff
@@ -3077,6 +3083,29 @@ private IEdmStringConstantExpression GetDescriptionAnnotation(IEdmVocabularyAnno
         return model.VocabularyAnnotations(this.context.EdmModel).Where(x => x.Term.Name == "Description")
             .Select(x => x.Value).FirstOrDefault() as IEdmStringConstantExpression;
     }
+
+    /// <summary>
+    /// Searches through model's vocabulary annotations and returns annotation of term 'Revisions' if present
+    /// </summary>
+    private IEdmStringConstantExpression GetRevisionsAnnotation(IEdmVocabularyAnnotatable model)
+    {
+        IEnumerable<IEdmExpression> collection = model.VocabularyAnnotations(this.context.EdmModel).Where(x => x.Term.Name == "Revisions")
+            .Select(x => x.Value);
+
+        if (collection != null && collection.Count() > 0)
+        {
+            IEdmCollectionExpression semanticElement = collection.FirstOrDefault() as IEdmCollectionExpression;
+            IEdmRecordExpression element = semanticElement.Elements.FirstOrDefault() as IEdmRecordExpression;
```

Comment:
```suggestion
            IEdmRecordExpression element = semanticElement?.Elements.FirstOrDefault() as IEdmRecordExpression;
```

Since you're using `FirstOrDefault()`, I think you should check for nulls, just in case.